src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java

```diff
@@ -71,7 +122,88 @@ private void submitRecordDeleteOperation() {}
 
   private void submitRecordReadOperation() {}
 
-  private void submitRecordCreateOperation() {}
+  boolean completeFailedOperation(final O op, final Throwable thrown) {
+    thrown.printStackTrace();
+    return completeOperation(op, FAIL_UNKNOWN);
+  }
+
+  boolean completeOperation(final O op, final Operation.Status status) {
+    concurrencyThrottle.release();
+    op.status(status);
+    op.finishRequest();
+    op.startResponse();
+    op.finishResponse();
+    return handleCompleted(op);
+  }
+
+  private Properties createAdminConfig(String nodeAddr) {
+    var adminConfig = new Properties();
+    adminConfig.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, nodeAddr);
+    adminConfig.put(AdminClientConfig.SEND_BUFFER_CONFIG, this.sndBuf);
+    adminConfig.put(AdminClientConfig.RECEIVE_BUFFER_CONFIG, this.rcvBuf);
+    return adminConfig;
+  }
+
+  private Properties createProducerConfig(String nodeAddr) {
+    var producerConfig = new Properties();
+    producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, nodeAddr);
+    producerConfig.put(ProducerConfig.BATCH_SIZE_CONFIG, this.batch);
+    producerConfig.put(ProducerConfig.MAX_REQUEST_SIZE_CONFIG, this.request);
+    producerConfig.put(ProducerConfig.BUFFER_MEMORY_CONFIG, this.buffer);
+    producerConfig.put(ProducerConfig.COMPRESSION_TYPE_CONFIG, compression);
+    producerConfig.put(ProducerConfig.SEND_BUFFER_CONFIG, this.sndBuf);
+    producerConfig.put(ProducerConfig.LINGER_MS_CONFIG, this.linger);
+    producerConfig.put(ProducerConfig.RECEIVE_BUFFER_CONFIG, this.rcvBuf);
+    producerConfig.put(
+        ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+        "org.apache.kafka.common.serialization.StringSerializer");
+    producerConfig.put(
+        ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+        "com.emc.mongoose.storage.driver.kafka.io.DataItemSerializer");
+    return producerConfig;
+  }
+
+  private void submitRecordCreateOperation(final DataOperation recordOp, String nodeAddr) {
+    try {
+      var adminConfig =
```

Comment:
1. Let's use the config cache in order to not create the config on each invocation
2. You can use single producer config instead of admin config too
3. Use `val` instead of `var` for the immutable values. This brings in a bit more safety